### PR TITLE
controller: Add and use lookup function for controllers

### PR DESCRIPTION
**Pull request by christarazi**
This commit has no functional impact, but eases future changes in this
code to reuse the lookup function.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

### CODE REVIEW

File: pkg/controller/manager.go [hunk 1]
```diff
@@ -161,8 +163,8 @@ func (m *Manager) removeAndReturnController(name string) (*managedController, er
 		return nil, fmt.Errorf("empty controller map")
 	}
 
-	oldCtrl, ok := m.controllers[name]
-	if !ok {
+	oldCtrl := m.lookupLocked(name)
```
Comment (ccfishk) [inline pkg/controller/manager.go:166]: From concurrency point of view, there is no different from the delete line 164, 165.
Is this the intention ?

Comment (christarazi) [reply pkg/controller/manager.go:166]: Yes, there should not be any behavioral difference.

**Comment by christarazi** [comment]
@danehans Could you take a look here?

**Review by pippolo84 — APPROVED** [review]
:rocket:

**Comment by danehans** [comment]
@christarazi sorry for not getting to this review. I was at KubeCon this week and had to prepare for two talks.

**Pull request merged**